.autorc.json

```diff
@@ -0,0 +1,9 @@
+{
+  "plugins": [
+    "npm",
+    "conventional-commits",
+    "first-time-contributor",
+    "released"
+  ],
+  "onlyPublishWithReleaseLabel": true
```

Comment:
@domoritz , if we remove this boolean, the library will auto-publish on every merge to `master`. I suggest we don't enable that until you're first happy with how the auto-version incrementing logic works (it depends on committers adhering well to git commit message guidelines). 

There are other options for setting version (setting a major/minor/patch github label), or asking users to check off the appropriate checkbox in their PR description when opening the PR, but this is probably the easiest place to start.

Reply:
I was wondering about all the labels that this process added. I am pretty strict with enforcing correct pull request descriptions withs semantic commits. Can we use them to automatically set the right version? Or is the idea of the labels that they get added automatically?

Reply:
> Can we use them to automatically set the right version?

I wasn't sure about the semantic commit situation in this repo because I didn't see commit-hooks. However, I think I saw some linting config in a different vega repo. Whether having good commit messages is checked by eye or tool, https://github.com/nteract/data-explorer/pull/66, having manual labeling as an option is just a fallback for inaccurate messages. 

The automatic labeling is based on `conventional-commits`, which most likely has the same behavior as the semantic-release plugin.

https://intuit.github.io/auto/docs/generated/conventional-commits

If you wanted to do the manual checkbox thing, it could be achieved with this plugin:

https://intuit.github.io/auto/docs/generated/pr-body-labels

Reply:
Last link: if any projects have more specific labeling needs than the defaults (or different mappings in terms of what we consider major/minor), they can be customized by passing in a custom array in the config:

https://intuit.github.io/auto/docs/configuration/autorc#labels

Reply:
Here is how we ensure commit messages:

<img width="912" alt="Screen Shot 2021-07-15 at 21 34 53" src="https://user-images.githubusercontent.com/589034/125892338-703f273b-3255-4b65-ada6-a1f2ff512f8c.png">

I don't want additional labels if they are not necessary.

Reply:
Oh neat I missed the `semantic pull request` integration! That's even more reliable than a commit hook since it doesn't depend on someone setting up local config properly. I might set that up for another project, thanks for the tip.

Regarding labels: I *think* we can delete the unused tags then. The other ones are optional and by default would help organize the auto-generated changelog, but are not required. I just deleted most of the added tags, with the exception of `release` and `released` since those are important for keeping track of what has been deployed. 

- https://intuit.github.io/auto/docs/welcome/getting-started#3-create-labels